src/WidgetBase.ts

```diff
@@ -385,6 +391,13 @@ export class WidgetBase<P extends WidgetProperties> extends Evented implements W
 			const { children, properties = {} } = dNode;
 			const { key } = properties;
 
+			const propagatedPropertyNames = this.getDecorator('propagateProperty');
```

Comment:
Yeah we could add a `PropagatePropertiesMixin`, but using the decorator wouldn't be so convenient as you'd have to make sure you used this mixin (and know that something else you're using, like `ThemeableMixin` is already using it).